pkg/frontend/querymiddleware/split_and_cache.go

```diff
@@ -143,18 +146,26 @@ func (s *splitAndCacheMiddleware) Do(ctx context.Context, req MetricsQueryReques
 		return nil, apierror.New(apierror.TypeBadData, err.Error())
 	}
 
+	isCacheEnabled := s.cacheEnabled && (s.shouldCacheReq == nil || s.shouldCacheReq(req))
+	maxCacheFreshness := validation.MaxDurationPerTenant(tenantIDs, s.limits.MaxCacheFreshness)
+	maxCacheTime := int64(model.Now().Add(-maxCacheFreshness))
+	cacheUnalignedRequests := validation.AllTrueBooleansPerTenant(tenantIDs, s.limits.ResultsCacheForUnalignedQueryEnabled)
+
+	// Force queier to track PerStepStats, so they could be cached.
+	if s.cacheSamplesProcessedStats {
```

Comment:
i'm not sure if this shouldn't be called something like `computeSamplesProcessedStats` - it controls whether we compute the stats at all, not only whether we don't cache them. To me cache means that the thing we're caching already exists, we just choose to store it in the cache. 

nitpick i guess

Reply:
I'm not fully happu with name, but let me give my perspective - w anyway computing Sample Processed, so it's more like `computeSamplesProcessedPerStep`. On the other hand, we compute them only to count SamplesProcessed from cache correctly, so that's why I ended up with cacheSamplesProcessedStats - to highlight feature, not the implementation